### PR TITLE
OOP with commands

### PR DESCRIPTION
**Pull request by duncte123**
The reason to use OOP for this is that the commands look cleaner when we use classes compared to when we export an object.

### CODE REVIEW

**Comment by duncte123** [comment]
> is BaseCommand really necessary though, it's mostly rewriting it anyway?

I would say that it is since it gives some validation against using the wrong run method

File: commands/base/BaseCommand.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+export default class BaseCommand {
+  #name = null;
+  #category = null;
+  #description = null;
+
+  constructor(options) {
+    if (!options.name || !options.category || !options.description) {
+      throw new Error("One of the configuration options is missing");
```
Comment (iamdeja) [inline commands/base/BaseCommand.js]: Unhandled?

File: commands/base/BaseCommand.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    this.#name = options.name;
+    this.#category = options.category;
+    this.#description = options.description;
+  }
+
+  run(client, message, args) {
+    throw new Error(`Run function not overwritten in ${this.constructor.name}`);
```
Comment (iamdeja) [inline commands/base/BaseCommand.js]: Unhandled?

File: commands/Law.js [hunk 3]
```diff
@@ -1,11 +1,17 @@
...
+  constructor() {
+    super({
+      name: "law",
+      category: "info",
+      description: "States the buddy law."
+    });
+  }
+
+  run(client, message) {
+    const embed = new Discord.MessageEmbed()
```
Comment (iamdeja) [inline commands/Law.js]: Import the message embed from Discord, it's cleaner

**Comment by duncte123** [comment]
Please merge #10 before this one

**Comment by not-archiey** [comment]
Looks less cluttered~!

**Pull request merged**